clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -983,10 +983,14 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
 
   if (getLangOpts().SYCLIsDevice && D) {
     if (const auto *A = D->getAttr<SYCLIntelLoopFuseAttr>()) {
-      Expr *E = A->getValue();
+      const Expr *Arg = A->getValue();
+      assert(Arg && "Got an unexpected null argument");
```

Comment:
No need to add this assertion here, `dyn_cast<>` will assert on nullptr (there's a `dyn_cast_or_null<>` for that case).

Reply:
Done